p2p/ipld/read.go

```diff
@@ -85,6 +85,7 @@ func uniqueRandNumbers(count, max int) []uint32 {
 	}
 	samples := make(map[uint32]struct{}, count)
 	for i := 0; i < count; {
+		// nolint:gosec // G404: Use of weak random number generator
 		sample := uint32(rand.Intn(max))
```

Comment:
is this OK?

Reply:
Good question. I think in this case it doesn't really matter as we are downloading enough data to reconstruct the whole EDS anyways. For the sampling, we should use something "more random" IMO.

Reply:
My only concern around using a weak PRNG is that it could allow some weird timing attacks to correlate requests to nodes. Only really problematic for DAS rather than full downloading.